libs/x-content/src/main/java/org/elasticsearch/xcontent/Text.java

```diff
@@ -36,31 +31,35 @@ public static Text[] convertFromStringArray(String[] strings) {
         return texts;
     }
 
-    private BytesReference bytes;
+    private ByteBuffer bytes;
     private String text;
     private int hash;
+    private int length = -1;
+
+    public Text(ByteBuffer bytes) {
+        this.bytes = bytes;
+    }
 
-    public Text(BytesReference bytes) {
+    public Text(ByteBuffer bytes, int length) {
```

Comment:
Is this ctor ever used? It does not seem to me; I think it's better to remove it, also because there is a potential for misunderstanding here (length can be the bytes length or the string length, which are potentially different I think)

Reply:
Length here would be the string length. I added this constructor to use in my other PR, since there are several checks in `KeywordFieldMapper` that need to know the length of the string. If I use the `ByteBuffer`-only constructor, then this class will perform the string conversion when checking the string length. Since we're trying to avoid said string conversion, I need a way to specify the string length explicitly.

I'll update the name of the variable to `stringLength`, and add a javadoc for clarity.